packages/qdrant-loader/src/qdrant_loader/core/project_manager.py

```
@@ -193,10 +194,21 @@ async def _ensure_project_in_database(
+            await self._update_project_sources(session, context.project_id, config)
+            await session.commit()
+        except IntegrityError as e:
+            # The unique-constraint error can be triggered by autoflush during
+            # session.execute() inside _update_project_sources, before commit.
+            await session.rollback()
+            err = str(e)
+            if "projects.collection_name" in err or "uix_project_collection" in err:
+                raise ValueError(
+                    "State DB schema is outdated and still enforces unique projects.collection_name. "
+                    "Reset state DB once and retry: `qdrant-loader init --workspace . --force`."
+                ) from e
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)